src/sparsezoo/v2/objects/model_directory.py

```diff
@@ -186,15 +176,19 @@ def from_directory(cls, directory_path: str) -> "ModelDirectory":
         :return: ModelDirectory class object
         """
         files = []
-        paths = glob.glob(os.path.join(directory_path, "*"))
-        if not paths:
+        display_names = os.listdir(directory_path)
+        if not display_names:
             raise ValueError(
                 "The directory path is empty. "
                 "Check whether the indicated directory exists."
             )
-        for path in paths:
-            display_name = os.path.basename(path)
-            files.append(file_dictionary(display_name=display_name, path=path))
+        for display_name in display_names:
+            files.append(
+                file_dictionary(
```

Comment:
nit: This function is only used once, could we explicitly create the dictionary here?

Reply:
good idea!

Reply:
Actually, this factory method will be deprecated in the next PR, so let's drop this for now. But I will take note to use your suggestion in the other function which will use the same logic ok?